lib/Analysis/LevelAnalysis/LevelAnalysis.h

```diff
@@ -94,8 +94,29 @@ class LevelAnalysis
   }
 };
 
+class LevelAnalysisBackward
```

Comment:
Maybe a little code comment to state why you need a separate backwards analysis.